gke-deploy/core/gcs/gcs.go

```diff
@@ -0,0 +1,91 @@
+// Package gcs contains logic related to Google Cloud Storage.
+package gcs
+
+import (
+	"context"
+	"errors"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/services"
+)
+
+const (
+	timeoutErr = "GCS timeout"
+)
+
+var (
+	defaultTimeout = 1 * time.Hour
+	defaultDelay   = 2 * time.Second
+	errGCSTimeout  = errors.New(timeoutErr)
+)
+
+type GCS struct {
+	Timeout    time.Duration
+	Retries    int
+	Delay      time.Duration
+	GcsService services.GcsService
+}
+
+// Download copies file(s) from GCS. <dst> should be a directory, not a path to a file
+func (s *GCS) Download(ctx context.Context, src, dst string, recursive bool) error {
+	log.Printf("Downloading file(s) %s from %s.\n", src, dst)
+	return s.copyWithRetry(ctx, src, dst, recursive)
+}
+
+// Upload copies file(s) to GCS.
+func (s *GCS) Upload(ctx context.Context, src, dst string) error {
+	log.Printf("Uploading file(s) %s to %s.\n", src, dst)
+	return s.copyWithRetry(ctx, src, dst, false)
+
+}
+
+// copyWithRetry is responsible for trying (and retrying) to call copyWithTimeout()
+// with appropriate retry backoff.
+func (s *GCS) copyWithRetry(ctx context.Context, src, dst string, recursive bool) error {
+	var err error
+	delay := s.Delay
+	if int64(delay) == 0 {
```

Comment:
I think Brandon mentioned that you don't have to cast this to `int64` since its already an alias of `int64`: https://golang.org/pkg/time/#Duration